### DISCUSSION TITLE
Livewire v3 install problem in windows.

### DISCUSSION THREAD

**Discussion by sajjadhossainshohag** [livewire/livewire#5775] [7. Bugs (archived)] [answered]
I am trying to install Livewire v3 version. I get a case-sensitive error in Windows.

**Command**: composer require livewire/livewire:^3.0@beta

**Error**:  The archive may contain identical file names with different capitalization (which fails on case-insensitive filesystems): ZipArchive::extractTo(C:\la  
  ragon\www\livewire/vendor/composer/ac7cb080/livewire-livewire-ea97977\js\directives/wire:dirty.js): Operation failed: No such file or directory.

Screenshot in below:

![Screenshot 2023-07-21 001435](https://github.com/livewire/livewire/assets/63788037/a58d5c1a-13a8-4f8b-ac29-935684d7ce5b)

**Comment by mcpuishor** [comment]
After all the wait and excitement, I tried to install with the same command

`composer require livewire/livewire:^3.0@beta`

I get under Windows 11, running a WAMP server with PHP 8.2  a different error, screenshot below. 

However, when I try the strict version: `composer require livewire/livewire:3.0.0-beta.1` I get the error screenshot above. 

![image](https://github.com/livewire/livewire/assets/40750444/91241bbd-5b77-4a2c-8fe6-8105e6707358)

**Reply by gixroq** [reply]
Got same error on windows, but I installed livewire v2 then upgraded to v3. That worked.

**Comment by akunbeben** [comment]
![image](https://github.com/livewire/livewire/assets/46495960/925a1fe0-1993-4d15-9aa7-1251b5627f9d)

I having the same issue, on Mac Monterey 12.6.6

```
The archive may contain identical file names with different capitalization (which fails on case insensitive filesystems)
    Unzip with unzip command failed, falling back to ZipArchive class
```

I thought it was my composer problem. I did move from brew's composer to the newest official app "Herd", still the same.

**Reply by akunbeben** [reply]
UPDATE: On the mac, it's a disk case sensitivity issue. 

I've created a new volume and enabled the case-sensitive. Issue solved.

**Reply by joshhanley** [reply]
@akunbeben yeah the merge failed and we end up with duplicates filenames with different cases. We will be tagging a release soon that fixes that so you don't need to change your mac volumes 😅

**Comment by joshhanley** [comment, marked as answer]
There are two separate issues here.

One is the Windows issue that @sajjadhossainshohagbd and @mcpuishor have, I have submitted a PR #5791 with a fix for that.

Then there is a file conflict issue that @akunbeben is seeing, I have submitted a PR #5789 with a fix for that.

Hope this helps!

**Reply by akunbeben** [reply]
Thanks for your response, I've solved my issue now.

https://github.com/livewire/livewire/discussions/5775#discussioncomment-6503671

**Comment by joshhanley** [comment]
@sajjadhossainshohagbd @mcpuishor @akunbeben I have tagged a new version beta.2 with the fixes, can you test and let me know if it fixes it? Thanks!

**Reply by wesleyamaro** [reply]
This solved my problem! Thank you very much!

**Reply by Dev-RaheelAhmad** [reply]
@joshhanley Bundle of Thanks! 🙏 `composer require livewire/livewire:^3.0.0-beta.2`

**Reply by joshhanley** [reply]
@wesleyamaro @Dev-RaheelAhmad awesome! thanks for confirming 🙂 You can also just do `composer require livewire/livewire:^3.0@beta` then when you run `composer update` it will pull the latest beta version.

**Reply by akunbeben** [reply]
Thanks Josh!

**Reply by Dev-RaheelAhmad** [reply]
@joshhanley `composer require livewire/livewire:^3.0@beta` is still giving the same error that's why I mentioned `composer require livewire/livewire:^3.0.0-beta.2`

composer require livewire/livewire:^3.0@beta
./composer.json has been updated
Running composer update livewire/livewire
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires livewire/livewire 3.0@beta, found livewire/livewire[dev-throw-error-when-testing-non-livewire-class, ..., dev-bad-hotfix-test, v0.0.1, ..., v0.7.4, v1.0.0, ..., 1.x-dev, v2.0.0, ..., 2.x-dev, v3.0.0-beta.1, v3.0.0-beta.2] but it does not match the constraint.

Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions.

Installation failed, reverting ./composer.json and ./composer.lock to their original content.

**Reply by joshhanley** [reply]
@Dev-RaheelAhmad ah interesting I wonder why it's doing that? Are you on windows? Works for me on Mac.

**Reply by Dev-RaheelAhmad** [reply]
Yes I'm on Windows 10 using XAMPP. 
Another weird thing is that after I ran `composer require livewire/livewire:^3.0.0-beta.2`, it didn’t add the @livewireStyles and @livewireScripts directives automatically, also didn’t create the components and layouts folder... 😕

**Reply by sajjadhossainshohag** [reply]
@joshhanley Thanks for your response, now works fine.

**Reply by mcpuishor** [reply]
All working well now. 
Thanks for your effort @joshhanley

**Reply by sfc3lf44** [reply]
I have the same issue on windows with XAMPP.  I was able to get beta.2 installed correctly, (which fixed the first error.  Thank you!) but composer require livewire/livewire:^3.0@beta gives the same error as Dev-RahellAhmad described.
 Problem 1
    - Root composer.json requires livewire/livewire 3.0@beta, found livewire/livewire[dev-throw-error-when-testing-non-livewire-class, ..., dev-bad-hotfix-test, v0.0.1, ..., v0.7.4, v1.0.0, ..., 1.x-dev, v2.0.0, ..., 2.x-dev, v3.0.0-beta.1, v3.0.0-beta.2] but it does not match the constraint.

Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions.

**Reply by joshhanley** [reply]
No worries! 🙂

@Dev-RaheelAhmad and @sfc3lf44 can you try this instead (using quote marks) `composer require "livewire/livewire:^3.0@beta"` and let me know if it works?

**Reply by sfc3lf44** [reply]
No luck this time.  This was the error message:
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires livewire/livewire 3.0@beta, found livewire/livewire[dev-throw-error-when-testing-non-livewire-class, ..., dev-bad-hotfix-test, v0.0.1, ..., v0.7.4, v1.0.0, ..., 1.x-dev, v2.0.0, ..., 2.x-dev, v3.0.0-beta.1, v3.0.0-beta.2, v3.0.0-beta.3] but it does not match the constraint.

Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions.

Installation failed, reverting ./composer.json and ./composer.lock to their original content.

**Reply by Dev-RaheelAhmad** [reply]
@joshhanley `composer require "livewire/livewire:^3.0@beta"` works without errors. I don’t know if the quotes matter or if it’s because I ran `php artisan livewire:upgrade` yesterday, which fixed the issue.

**Reply by joshhanley** [reply]
@Dev-RaheelAhmad yeah sometimes terminals don't like things like the `@` symbol so putting in quotes can help.

@sfc3lf44 hmm not sure why yours is still failing but you do have it installed using the specific beta version yeah?

**Reply by Dev-RaheelAhmad** [reply]
@joshhanley could you please guide me on how to report the typos or errors in the new v3 docs? I have noticed some issues in the docs but not sure how and where to share them.

**Reply by joshhanley** [reply]
@Dev-RaheelAhmad yep! All docs markdown files are now in the main repo in the `/docs` directory, so you can just submit a pull request to `livewire/livewire` with the changes. 🙂

**Comment by samharvey44** [comment]
Just to add, I had the same issue, but surrounding in quotes fixed the problem (`composer require "livewire/livewire:^3.0@beta" ` rather than `composer require livewire/livewire:^3.0@beta`)

**Reply by joshhanley** [reply]
@samharvey44 great! Thanks for confirming

**Comment by inmoh7** [comment]
These issues happened to me while using the terminal inside the VS code editor (which is in my case default to power-shell). Both 
- composer require livewire/livewire "^3.0@beta"
- composer require "livewire/livewire:^3.0@beta" 

didn't work. Then I switched the terminal to CMD and worked perfectly fine.

**Reply by joshhanley** [reply]
@Infas7 please see our warning about using PowerShell  in the installation guide.

<img width="767" alt="image" src="https://github.com/livewire/livewire/assets/882837/bd4084be-27e3-472d-ace7-c92ccf214d99">

**Reply by Dev-RaheelAhmad** [reply]
@joshhanley sorry for mentioning here, it's related to different topic but could you please help me 🙏 https://github.com/livewire/livewire/discussions/6290 https://github.com/livewire/livewire/discussions/6253 
Even I mentioned this as a bug: https://github.com/livewire/livewire/discussions/6422
